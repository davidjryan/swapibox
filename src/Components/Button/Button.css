button {
  border-radius: 4px;
  font-size: 16px;
  padding: 10px;
  margin: 10px 20px; }

.favorite-button {
  background-color: #EDFFF0;
  color: #3F0E10;
  background: url(../../Assets/Rebel_Alliance_logo_black.svg);
  background-size: cover;
  height: 160px;
  width: 160px;
  border-style: none;
  display: inline-block; }

.favorite-button:hover {
  background: url(../../Assets/Rebel_Alliance_logo_white.svg);
  background-repeat: none;
  background-size: cover;
  background-color: none; }

.nav-btn {
  background-color: #191A21;
  color: #e5b13a;
  border: 2px solid #e5b13a; }

.nav-btn:hover, .activeBtn {
  background-color: #e5b13a;
  color: #191A21; }

.activeBtn {
  font-size: 1.3em;
  font-weight: bold;
  border: none; }
