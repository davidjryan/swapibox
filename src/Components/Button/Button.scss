@import "../../variables.scss";

button {
  border-radius : 4px;
  font-size: 16px;
  padding: 10px;
  margin: 10px 20px;
}

.favorite-button {
  background-color: $text-light;
  color: $body-dark;
  background: url(../../Assets/Rebel_Alliance_logo_black.svg);
  background-size: cover;
  height: 160px;
  width: 160px;
  border-style: none;
  display: inline-block;
}

.favorite-button:hover {
  background: url(../../Assets/Rebel_Alliance_logo_white.svg);
  background-repeat: none;
  background-size: cover;
  background-color: none;
}

.nav-btn {
  background-color: $text-dark;
  color: $crawl-color;
  border: 2px solid $crawl-color;
}

.nav-btn:hover, .activeBtn {
  background-color: $crawl-color;
  color: $text-dark;
}

.activeBtn {
  font-size: 1.3em;
  font-weight: bold;
  border: none;
}

.favorites-btn {

}
