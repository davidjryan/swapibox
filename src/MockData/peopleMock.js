module.exports = [

    {
      Name: "Luke Skywalker",
      Species: "Human",
      Language: "Galactic Basic",
      "Homeworld Population": "200000",
      Homeworld: "Tatooine",
      favorite: false
    },
    {
      Name: "C-3PO",
      Species: "Droid",
      Language: "n/a",
      "Homeworld Population": "200000",
      Homeworld: "Tatooine",
      favorite: false
    },
    {
      Name: "R2-D2",
      Species: "Droid",
      Language: "n/a",
      "Homeworld Population": "4500000000",
      Homeworld: "Naboo",
      favorite: false
    }

];
