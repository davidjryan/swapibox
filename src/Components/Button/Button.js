import React from 'react';
import './Button.css';
import PropTypes from 'prop-types';

const Button = ({ click, text, givenClass, category }) => {
  let buttonClass = `${givenClass} nav-btn`

  return (
    <button onClick={() => click(category)}
            className={buttonClass}>
      {text}
      {/* {favNum} */}
    </button>
  )
}

// Button.propTypes = {
//   click: PropTypes.func.isRequired,
//   text: PropTypes.string,
//   favNum: PropTypes.number,
//   setClass: PropTypes.string,
//   card: PropTypes.object
// }

export default Button
