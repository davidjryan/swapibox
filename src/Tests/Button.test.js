import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Button from '../Components/Button/Button';
import peopleMock from '../MockData/peopleMock';

describe('Button', () => {

  it('should create an instance of a button', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<Button
      givenClass='favorite-button nav-btn'
      test=""
      click={mockFunc}
      category={peopleMock[0]}
      cardInformation={peopleMock[0]}
    />);

    expect(wrapper).toMatchSnapshot();
  });
})
