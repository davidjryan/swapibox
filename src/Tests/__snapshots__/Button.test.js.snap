// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should create an instance of a button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    cardInformation={
        Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "Galactic Basic",
            "Name": "Luke Skywalker",
            "Species": "Human",
            "favorite": false,
          }
    }
    category={
        Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "Galactic Basic",
            "Name": "Luke Skywalker",
            "Species": "Human",
            "favorite": false,
          }
    }
    click={[Function]}
    givenClass="favorite-button nav-btn"
    test=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "favorite-button nav-btn",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "favorite-button nav-btn",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
